test/fail_compilation/retscope.d

```diff
@@ -302,3 +302,28 @@ struct S7
 
 int[3] escape8(scope int[] p) @safe { return p[0 .. 3]; } // should not error
 char*[3] escape9(scope char*[] p) @safe { return p[0 .. 3]; }
+
+/***************************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(320): Error: reference to local variable i assigned to non-scope f
+---
+*/
+
+int* escape10() @safe
+{
+    int i;
+    int* f;
+    scope int** x = &f;
```

Comment:
A bit surprising that this doesn't work, because `lifetime(x) < lifetime(f)`. Also `int i; scope int* p = &i;` on it's own does work, doesn't it?

If `lifetime(*x)` simply was defined as `lifetime(x)`, there wouldn't be a way to escape the pointer to f, but I'm not sure how much DIP1000 still relates to any of the implementation at this point.